files-csharp/CountFilesInaFolderUsingCSharp/CountFilesInaFolderTests/CountFilesInaFolderTests.csproj

```diff
@@ -0,0 +1,37 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net8.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <IsPackable>false</IsPackable>
+    <IsTestProject>true</IsTestProject>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
+    <DefineConstants>$(DefineConstants);WINDOWS</DefineConstants>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <DefineConstants>$(DefineConstants);WINDOWS</DefineConstants>
+  </PropertyGroup>
```

Comment:
I don't think this constant define is necessary. The code will build just fine. We just need to check the platform (https://learn.microsoft.com/en-us/dotnet/api/system.operatingsystem.iswindows?view=net-8.0) before trying to execute the windows code. We could also use that check in the unit test itself to just pass the test if it is not on windows.

Reply:
I originally defined them to allow me to use the preprocessor directives to conditionally compile the `GivenDirectoryWithFiles_WhenUsingWinAPI_ThenReturnCorrectFileCount()` unit test that is now commented out because it was failing to pass the Jenkins CI build. But I'll get rid of it.

Reply:
This is all you need to do to prevent the unit test failing when running on non-Windows OSes:
![image](https://github.com/CodeMazeBlog/CodeMazeGuides/assets/1489638/cda3a583-c7a6-47ef-bfc6-75e24e06a226)